include/openmc/settings.h

```diff
@@ -109,6 +118,10 @@ extern array<double, 4>
   energy_cutoff; //!< Energy cutoff in [eV] for each particle type
 extern array<double, 4>
   time_cutoff; //!< Time cutoff in [s] for each particle type
+extern int
+  ifp_n_generation; //!< Number of generation for Iterated Fission Probability
+extern IFPParameter
+  ifp_parameter; //!< Parameter to calculate for Iterated Fission Probability
```

Comment:
I see right now this is determined based on user input in the `Settings` object. I think it would be better if this was determined automatically based on what scores show up in tallies (i.e., if the user specified "ifp-time-numerator" as a score, then activate the logic to collect the information needed accordingly), which leaves less for the user to think about.

Reply:
The code now detects automatically the correct value for ifp_parameter based on the user defined tallies. It also detects whether settings::ifp should be true or not. I removed the iterated_fission_probability settings so that the user only declares the number of generations via ifp_n_generation instead. If the user does not declare this value and IFP tallies are requested, the code will use a default value of 10 generations.